Classes/ViewHelpers/Uri/AbstractUriViewHelper.php

```diff
@@ -40,21 +41,31 @@ abstract class AbstractUriViewHelper extends AbstractSolrFrontendViewHelper
 
     protected static SearchUriBuilder $searchUriBuilder;
 
+    protected static RequestBuilder $requestBuilder;
+
     public function injectSearchUriBuilder(SearchUriBuilder $searchUriBuilder): void
     {
         self::$searchUriBuilder = $searchUriBuilder;
     }
 
+    public function injectRequestBuilder(RequestBuilder $requestBuilder): void
+    {
+        self::$requestBuilder = $requestBuilder;
+    }
+
     protected static function getSearchUriBuilder(RenderingContextInterface $renderingContext = null): SearchUriBuilder
     {
         if (!isset(self::$searchUriBuilder)) {
             self::$searchUriBuilder = GeneralUtility::makeInstance(SearchUriBuilder::class);
         }
+        if (!isset(self::$requestBuilder)) {
+            self::$requestBuilder = GeneralUtility::makeInstance(RequestBuilder::class);
+        }
 
         if ($renderingContext instanceof RenderingContext) {
             $uriBuilder = GeneralUtility::makeInstance(UriBuilder::class);
-            /** @phpstan-ignore-next-line */
-            $uriBuilder->reset()->setRequest($renderingContext->getRequest());
+            $request = self::$requestBuilder->build($renderingContext->getRequest());
```

Comment:
Maybe we should check whether the Request of Rendering Context already is an extbase request and keep original logic in that case. Only use new one in case we didn't receive an extbase request?

Reply:
Its ok so.